spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkCachedTableCatalog.java

```diff
@@ -144,12 +151,25 @@ private Pair<Table, Long> load(Identifier ident) throws NoSuchTableException {
       if (snapshotBasedMatcher.matches()) {
         snapshotId = Long.parseLong(snapshotBasedMatcher.group(1));
       }
+
+      Matcher branchBasedMatcher = BRANCH.matcher(meta);
+      if (branchBasedMatcher.matches()) {
+        branch = branchBasedMatcher.group(1);
```

Comment:
Question: I see we had a `continue` statement after the first match to avoid checking other branches. Does it make sense to repeat the same for new branches except the last one?

Reply:
nice catch!